lib/web_ui/lib/src/engine/util.dart

```diff
@@ -172,44 +156,16 @@ bool isIdentityFloat64ListTransform(Float64List matrix) {
   return transformKindOf(matrix) == TransformKind.identity;
 }
 
-/// Converts [matrix] to CSS transform value.
+/// Converts [matrix] to CSS transform 2D matrix value.
+///
+/// The [matrix] must not be a complex transforma, but one of 2D-only
+/// transforms.
 String float64ListToCssTransform2d(Float64List matrix) {
-  assert (transformKindOf(matrix) == TransformKind.scaleAndTranslate2d);
-  return 'matrix(${matrix[0]},0,0,${matrix[5]},${matrix[12]},${matrix[13]})';
-}
-
-/// Converts [matrix] to CSS transform value.
-String float64ListToCssTransform(Float64List matrix) {
-  assert(matrix.length == 16);
-  final Float64List m = matrix;
-  if (m[1] == 0.0 &&
-      m[2] == 0.0 &&
-      m[3] == 0.0 &&
-      m[4] == 0.0 &&
-      m[6] == 0.0 &&
-      m[7] == 0.0 &&
-      m[8] == 0.0 &&
-      m[9] == 0.0 &&
-      m[10] == 1.0 &&
-      m[11] == 0.0 &&
-      // 12 can be anything (translation)
-      // 13 can be anything (translation)
-      m[14] == 0.0 &&
-      m[15] == 1.0) {
-    final double tx = m[12];
-    final double ty = m[13];
-    if (m[0] == 1.0 &&
-          m[5] == 1.0) {
-        return 'translate(${tx}px, ${ty}px)';
-    } else {
-      return 'matrix(${m[0]},0,0,${m[5]},${tx},${ty})';
-    }
-  } else {
-    return 'matrix3d(${m[0]},${m[1]},${m[2]},${m[3]},${m[4]},${m[5]},${m[6]},${m[7]},${m[8]},${m[9]},${m[10]},${m[11]},${m[12]},${m[13]},${m[14]},${m[15]})';
-  }
+  assert (transformKindOf(matrix) != TransformKind.complex);
```

Comment:
Do we also wanna check that it's not a `TransformKind.identity`?

Reply:
It's not incorrect to pass an identity transform to this function. We assert that it's not `complex` because that would lead to visibly incorrect painting. However, I updated the docs, explaining that it's inefficient to construct the identity matrix unnecessarily.